### DISCUSSION TITLE
read value from a text file stored in storage?

### DISCUSSION THREAD

**Discussion by tryggs** [supabase/supabase#7713] [Questions]
I think my title explains my question but ask me if you don't understand what I am looking for!

Thanks in advance!

**Comment by GaryAustin1** [comment]
There is no way access storage data with out downloading the file.

**Reply by tryggs** [reply]
so there is no way to access the data with the API?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/reference/javascript/storage-from-download is the only API method to get file (filedata) besides using url's.

**Reply by tryggs** [reply]
all I am getting from the API is the file size and type. so is it possible to get the value of the file with the URL?

**Reply by GaryAustin1** [reply]
I get this in the response from .download() in js
![image](https://user-images.githubusercontent.com/54564956/179320807-d2100849-20f5-4538-89f8-c87fd6fc35eb.png)
That is a download of a csv file and the last line is from await data.text() (a function provided for a blob in js)

**Reply by tryggs** [reply]
thank you so much for your help! i figured it out now, I guess that the answer I needed was the text() function.
